app/javascript/components/gtl-view.jsx

```diff
@@ -181,7 +181,11 @@ const unSelectAll = (state) => {
 const gtlReducer = (state, action) => {
   switch (action.type) {
     case 'dataLoaded':
-      ManageIQ.gridChecks = [];
+      if (state && state.additionalOptions && state.additionalOptions.no_checkboxes_clicked
+        && state.additionalOptions.no_checkboxes_clicked.length === 0) {
+        // Making selected checkboxes array empty when those rows are  deleted or on compare/drift cancel
+        ManageIQ.gridChecks = [];
```

Comment:
Is ManageIQ.gridChecks effectively a global array?  If so, what happens if you check a bunch of VMs then go over to Hosts? (I do realize that gridChecks was there previously)

Reply:
Yeah it is essentially a global array. In that specific case ManageIQ.gridChecks does reset all its values, so there isn't any lingering values from previous iterations of the DataTable.